### ISSUE TITLE
Behavior is undeterministic if you reference a quality profile twice in the configuration

### ISSUE THREAD

**Issue by tmchow** [recyclarr/recyclarr#122]
Suppose you have this configuration. Based on what I've read in discord, this isn't supported.  I could see wanting to do this to specific different scores (overriding the default scores for a custom format):

```
    custom_formats:
      - trash_ids:
          - eca37840c13c6ef2dd0262b141a5482f # 4K remaster
          - 570bc9ebecd92723d2d21500f4be314c # Remaster
          - 0f12c086e289cf966fa5948eac571f44 # hybrid
          - ae43b294509409a6a13919dedd4764c4 # Repack2
          - e7718d7a3ce595f289bfee26adc178f5 # Repack/Proper
        quality_profiles:
          - name: Foobar
            score: 1000
            reset_unmatched_scores: true
      - trash_ids:
          - e23edd2482476e595fb990b12e7c609c # DV HDR10
        quality_profiles:
          - name: Foobar
            score: 500
            reset_unmatched_scores: true
```

**Comment by rcdailey** [comment]
What is the actual behavior you observe? I'm less interested in the discord discussion. I imagine you read my thoughts about this situation discussed there, but in fairness I never actually verified in the code what will happen here.

I think it's a good idea to see what it actually does and then talk about what we actually *want* it to do.

**Comment by tmchow** [comment]
@rcdailey I'd propose this:

1. "Reset unmatched scores" is not per section, but applies to the entire configuration file.  
2. I'd expect new custom formats to be added to the quality profiles even if the quality profile is mentioned a second time like i have in the above example
3.  If the same custom format is duplicated in successive sections, either an error is thrown or the latest reference wins

**Comment by rcdailey** [comment]
You forgot to describe the behavior you observe using your example.

**Comment by rcdailey** [comment]
I was finally able to spend some time on this and do some deep analysis. Since this isn't a case I've ever tested or anticipated, I honestly wasn't sure what the behavior would be. The good news is I think the behavior is already really good. There's one weird nuance that I'll get to at the end.

The good news is that the scores will be assigned properly as you'd expect. The "Foobar" profile will have a score of 1000 for each of the top 5 CFs listed in that example, with the sixth CF (DV HDR10) set to a score of 500. I believe this is the behavior we want.

The nuance here is that when `reset_unmatched_scores` is specified for the same profile twice, *only the first occurrence is used*. In your example this doesn't matter because both times you set it to `true`. But imagine if you had something like this instead:

```yml
    custom_formats:
      - trash_ids:
          - ...
        quality_profiles:
          - name: Foobar
            score: 1000
            reset_unmatched_scores: false
      - trash_ids:
          - ...
        quality_profiles:
          - name: Foobar
            score: 500
            reset_unmatched_scores: true
```

In this case, `reset_unmatched_scores: false` is the first occurrence, so this is the configuration that will be used. `reset_unmatched_scores: true` will be ignored, even though it has a different value.

I think this behavior is *also* acceptable, since it's difficult to reason about other ways to make this behavior different but useful for the same quality profile.

So in summary, Recyclarr makes sure to additively collect information you specify about CFs and manual scoring, no matter how you split it across objects in the `recyclarr.yml` file.

**Comment by rcdailey** [comment]
I almost forgot to respond to your last comment @tmchow. Please see below:

> "Reset unmatched scores" is not per section, but applies to the entire configuration file.

I don't want to do it this way because then people can't opt-out of this on a per-profile basis. I may want to reset scores in profile `A` but not profile `B`. I'd like to keep this flexibility.

> I'd expect new custom formats to be added to the quality profiles even if the quality profile is mentioned a second time like i have in the above example

As mentioned in my previous reply, this is in fact the current behavior.

> If the same custom format is duplicated in successive sections, either an error is thrown or the latest reference wins

Specifying the same CF multiple times for the same profile is not something I currently handle. I will certainly add logic for this. I think printing a warning and going with the first occurrence is appropriate.

**Comment by rcdailey** [comment]
For the following YAML (where DV HDR10 is duplicated within the same profile):

```yml
    custom_formats:
      - trash_ids:
          - eca37840c13c6ef2dd0262b141a5482f # 4K remaster
          - 570bc9ebecd92723d2d21500f4be314c # Remaster
          - 0f12c086e289cf966fa5948eac571f44 # hybrid
          - ae43b294509409a6a13919dedd4764c4 # Repack2
          - e7718d7a3ce595f289bfee26adc178f5 # Repack/Proper
          - e23edd2482476e595fb990b12e7c609c # DV HDR10
        quality_profiles:
          - name: SD
            score: 1000
            reset_unmatched_scores: true
      - trash_ids:
          - e23edd2482476e595fb990b12e7c609c # DV HDR10
        quality_profiles:
          - name: SD
            score: 500
            reset_unmatched_scores: true
```

I've made a change where you will get the following log output:

```txt
[WRN] Custom format with trash ID e23edd2482476e595fb990b12e7c609c is duplicated 1 times in quality profile SD with the following scores: [500]
[WRN] When the same CF is specified multiple times with different scores in the same quality profile, only the score from the first occurrence is used. To resolve
the duplication warnings above, remove the duplicate trash IDs from your YAML config
```

Does this address your third point @tmchow? This will be available in the next release.
